python/cudf/cudf/tests/test_groupby.py

```diff
@@ -291,6 +291,14 @@ def foo(df):
     got = got_grpby.apply(foo)
     assert_groupby_results_equal(expect, got)
 
+    def foo_args(df, k):
```

Comment:
I would prefer to see this in a separate test function rather than combined with tests for functions that lack `*args`. Maybe the data generation can be shared via a helper function (a fixture could work but would be evaluated at collection time, which is undesirable). Similarly for the other tests.

Reply:
tbh I agree with you here. Let me rework these a bit.

Reply:
What do you think of doing it kind of like https://github.com/rapidsai/cudf/pull/10682/commits/7b27cc5b90d99ae03a66de201fd0306a27fce47d ?

Reply:
I'm happy with that. 👍